### ISSUE TITLE
Renaming a storage node doesn't update child node names

### ISSUE THREAD

**Issue by snorfalorpagus** [pywr/pywr#48]

**Comment by jetuk** [comment]
Make the name property defer to `parent` if not None?

**Comment by snorfalorpagus** [comment]
This creates a bit of a problem for keeping `Model.nodes` updated.

**Comment by jetuk** [comment]
`Model.nodes` could be a generator instead?

**Comment by snorfalorpagus** [comment]
Like this? Seems inefficient, but it's not going to be called all that often.
```
for node in self.graph.nodes():
    if node.name == name:
        return node
raise KeyError()
```

**Comment by jetuk** [comment]
Are we taking about `Model.node` or `Model.nodes` or both?

**Comment by snorfalorpagus** [comment]
Sorry, `Model.node`.

**Comment by jetuk** [comment]
Perhaps we should be more explicit about we treat child nodes? I.e. should they even be accessible in `Model.node`? Should a user fetch the parent and access the child through that?

**Comment by snorfalorpagus** [comment]
Maybe you're right. Perhaps we don't need to add them to `Model.node` at all.

**Comment by jetuk** [comment]
But they returned by `Model.nodes` which would be confusing? I.e. `Model.nodes != Model.node.values()`

**Comment by snorfalorpagus** [comment]
`Model.nodes` just defers to `Model.graph.nodes`. I'm not sure we should be exposing `Model.nodes` and `Model.edges` like this. Perhaps we remove them, rename `Model.node` to `Model.nodes` and make it behave like this:

```
reservoir = model.nodes['myreservoir']
# or
for node in model.nodes:
    pass
```

This would be similar to the way Fiona iterates over features in a shapefile:
```
feature = dataset[42]
# or
for feature in dataset:
    pass
```

If the user really wants to see the graph, they can look at `Model.graph` directly.

**Comment by jetuk** [comment]
And this would only contain parent level nodes? Sounds good to me.

**Comment by snorfalorpagus** [comment]
This needs a flag on the node class, similar to `Node.allow_isolated` in #85.

`Node.show_graph` ?
`Node.hidden` ?
`Node.invisible` ?
???

**Comment by jetuk** [comment]
Can't we use `if self.parent is not None`?

**Comment by snorfalorpagus** [comment]
Closed by #89.
